### PR TITLE
Do not override the `destroy` method, use `_destroy` instead

### PR DESCRIPTION
**Pull request by lpinca**
`transform.destroy()` is now available in `readable-stream`, so I think it makes sense to use it properly.

### CODE REVIEW

File: stream.js [hunk 1]
```diff
@@ -12,18 +12,13 @@ function buildProxy (options, socketWrite, socketEnd) {
...
 
-  proxy._destroyed = false
   proxy._write = socketWrite
   proxy._flush = socketEnd
 
-  proxy.destroy = function(err) {
-    if (this._destroyed) return
-      this._destroyed = true
-
+  proxy._destroy = function(err, cb) {
```
Comment (lpinca) [inline stream.js]: Using the default implementation with

```
self.push(null)
self.end()
```

is probably better but I didn't change it as there might be something I'm missing here.

File: stream.js [hunk 2]
```diff
@@ -155,7 +150,7 @@ function WebSocketStream(target, protocols, options) {
 
   function onmessage(event) {
     var data = event.data
-    if (data instanceof ArrayBuffer) data = Buffer.from(new Uint8Array(data))
+    if (data instanceof ArrayBuffer) data = Buffer.from(data)
```
Comment (mcollina) [inline stream.js]: is this change needed?

File: stream.js [hunk 3]
```diff
@@ -155,7 +150,7 @@ function WebSocketStream(target, protocols, options) {
 
   function onmessage(event) {
     var data = event.data
-    if (data instanceof ArrayBuffer) data = Buffer.from(new Uint8Array(data))
```
Comment (lpinca) [inline stream.js]: This change is unrelated and not needed but there is no need to instantiate a new `Uint8Array`. You can use the `ArrayBuffer` directly.

I can revert this change or move it to its own commit/PR if wanted.

Comment (mcollina) [reply stream.js]: no need.
